### PR TITLE
Merge 4.x -> 5.x

### PR DESCRIPTION
**Pull request by mshabunin**
My attempt to make regular merge :slightly_smiling_face: 

Companion PRs: opencv/opencv_contrib#3758 and opencv/opencv_extra#1183

PR list:
a03b81316782ae30038b288fd3568993fa1e3538 Merge pull request #25732 from dkurt:opencv_js_tests_update
8e9307074c7ce6f15a98d444bb2313a5c7627ecd Merge pull request #25738 from rlexmann:imgproc_fix_divSpectrums
2629688d117dbf23a744263d191d49c85813c406 Merge pull request #25706 from cudawarped:fix_cuda_first_python_dep
3ea7ed9ebf926d53e2813ce2bde585b208aae167 Merge pull request #25718 from savuor:rv/hwasan_flag
bdf986ee519b01b0f451d1a3e71866d5027c0a47 Merge pull request #25726 from chacha21:remap_relative_doc
cc6f85e1bab6942acf107e4cacf813347f0552c1 Merge pull request #25427 from MaximSmolskiy:make-finding-corner-neighbor-symmetrical-in-ChessBoardDetector-findQuadNeighbors
3282954c2ea8d5873b8d50aff09d7b4a43a64c38 Merge pull request #25723 from mshabunin:fix-ts-rng
3700f9e1e932ca1aaa15db70534cb78b6e44ce05 Merge pull request #25709 from dkurt:wrap_addLayer
bef5a8768004c3d9b5eeb742446afe7d4e3e8145 Merge pull request #25722 from AleksandrPanov:update_testSeveralBoardsWithCustomIds
adcb0703965d2835eb76d040ae0a5e0fa151566b Merge pull request #25307 from MaximMilashchenko:halrvv071
cbf3b1187d9192d64511c2af01fb2aa2310b069d Merge pull request #25720 from VadimLevin:dev/vlevin/floodFill-optional-mask
0d1ed49d2a1221994358eccbf90b4be2b8f70b67 Merge pull request #25703 from mshabunin:fix-imread-arg
a5d9c29b121c85968e10d523a6baae46c59e5e16 Merge pull request #25716 from alexlyulkov:al/mediandk-fix
92b588f30bd29751be6dcb08321b047731e9878f Merge pull request #25702 from asmorkalov:as/gapi_disable_steaming_again
d8f0838fa35b879e24196feafbf9dd1d683f0af0 Merge pull request #25701 from keanep:I25700
a7e53aa1846ada1c4de6986a0d02bd7c740d2eec Merge pull request #25671 from savuor:rv/arithm_extend_tests
1bd5ca1ebe3279d5492f95f6c080475ef03fd74d Merge pull request #25686 from Kumataro:fix25674
1db6a8a1f33786880706426313d72725f5f7bf99 Merge pull request #25665 from vrabaud:jacobian
98b8825031f19f47b1e33a9b9c062208f8d4acb5 Merge pull request #25613 from CNOCycle:tflite/ops
29f91a08d53cf46c2b9183a879c098695ca42c08 Merge pull request #25680 from mshabunin:fix-approx-2
08107957d87bfb4b0f43c337130821e011d9dcd3 Merge pull request #25682 from fengyuentau:calib3d/lapack_fix_calib3d
472eba324af899e192c30970fe8dda98332a0f63 Merge pull request #25673 from AleksandrPanov:fix_charuco_board_markers
dcce2b8b2412dedb35f1ba238ffe40d68019d3ac Merge pull request #25662 from ujjayant-kadian:uk/port-gapi-onnxrt-backend-into-v2-api
d7f04a9d33d9c7d7c4d829fe6a5e55955d158e3a Merge pull request #25660 from Abdurrahheem:ash/fix-slice-empty-input
f4ebf7c0a6b2ee845a395ae3dac59927eb862894 Merge pull request #25679 from asmorkalov:as/HAL_min_max_idx_fix
7e9ef4db86cd403a8817832c8e558ef82c17733b Merge pull request #25625 from fengyuentau:core/deploy_fix_lapack_ilp64
3e3ee106fb8ccd003aa2c9a943a2340b066537bc Merge pull request #25658 from asmorkalov:as/variadic_tuple
2624929ec66ae66333311d88ba26aff43675d78c Merge pull request #25672 from mshabunin:fix-approx
79af357cf44214abde34c2f1ac79916849d179ce Merge pull request #25668 from vrabaud:legacy
d9421ac148f3ad4878b3a75c58da58c24f19ed01 Merge pull request #25167 from plctlab:rvp_3rdparty
05e48605a0aea00d3a89b9ab5e25cdf89568aa28 Merge pull request #25412 from ZelboK:update-cudnn-to-9
b6593517c4d2cd9f3d9e176e82ce43e6d0085e3e Merge pull request #25647 from Kumataro:fix25646
c5976f78659a836cc6f747083f7ee5b465d2db18 Merge pull request #25641 from vrabaud:lsh
f1890e384b535160c850b4b788ee4e2e52133be4 Merge pull request #25650 from sturkmen72:libjpeg-turbo
1fa96b161ff2525f446d829db06b6eb59850cdc4 Merge pull request #25616 from savuor:rv/yuv_docs
d97df262f6d25971e05c19731179996041f9e07f Merge pull request #25623 from asmorkalov:as/jpegturbo_3.0.3
7b31cc731416c66993ca90636d31e9a6447f03e5 Merge pull request #24897 from JStech:fix-handeye
cc49aee04e36b01f3b775863b7b5c633f06b41b9 Merge pull request #25634 from savuor:rv/boxfilter_tests
b267f1791c6901addd97570a222995b93934ac6f Merge pull request #25633 from savuor:rv/rotate_tests
f17b122512326ca51b07c2ba23bb174b4273a6df Merge pull request #25638 from bebuch:patch-2
ba28b7d6dfe5d4927fc5db46afeffdc3ba46b05b Merge pull request #25636 from FantasqueX:use-left-arrow-1
28c3000f6229756590429a2104edbd1b734ff713 Merge pull request #25629 from arobase-che:fix-opencv.js-compilation-memory-init-file
f1aef892bfe7b30255050275fdb80d2ec108a9a4 Merge pull request #25619 from opencv-pushbot:gitee/alalek/update_ffmpeg_4.x
f5793d26b7f362f8accefbaa7a241b4a5e0e2c5b Merge pull request #25632 from bebuch:patch-1
de3eb0ec5f0e27b74ea070ebc34ac4901119f583 Merge pull request #25626 from asmorkalov:as/HAL_lut_size
4824354e4673a58d8620975ce98bb93d13cf7220 Merge pull request #25631 from asmorkalov:as/png_build_fix
16b5096ed079d34296aa22277766ee85f403b2de Merge pull request #25618 from asmorkalov:as/kleidicv_0.1.0
9f1559595bc476099e448d931a6744065152945a Merge pull request #25627 from vrabaud:tiff
f2c48e5e9787850840535de33527da3c3a9d0e6a Merge pull request #25622 from lackhole:remove-unicode-character
e765c9f9c84437b0d3d3457e7b7f4aa358b7e0c1 Merge pull request #25580 from sturkmen72:libpng_1_6_43
8393885a39dac1e650bf5d0aaff84c04ad8bcdd3 Merge pull request #25615 from asmorkalov:update_version_4.10.0-pre
49f80cb3c42a91a8baa966b38664bdae4296db22 Merge pull request #24804 from fengyuentau:fix_lapack_warnings
e52540162fe40a341fc468b90049139b3523a87d Merge pull request #25586 from hanliutong:rvv-64f
c71d4952733a0e1dd1f88ac87066c802f1119d97 Merge pull request #25611 from asmorkalov:as/openjpeg_2.5.2
23ba63b0bea0fbf71d3a38836ae92d9ee9d0b532 Merge pull request #25574 from savuor:rv/hal_sub8x32f
69af621ef63a508ab9d1abbfa259a6876622e71e Merge pull request #25506 from savuor:rv/hal_mul16
83e32c4d37a8ddee41e7ecf5c9b42c6caa970a1d Merge pull request #25511 from savuor:rv/hal_projectpoints
e05ad56f6e7fa6dfcb8bc9962db9234a093679e8 Merge pull request #24903 from cristidbr-adapta:feature-barcode-detector-parameters
6350bfbf79562f257b58843b31e88a3245ee31a4 Merge pull request #25564 from mshabunin:cleanup-imgproc-2
8aa129dce1d3cae5857db30c166b40e50751003b Merge pull request #25603 from asmorkalov:as/skip_vulkan_concat
5e7887897a0e74afe76f37b6604ac865e8016ddb Merge pull request #25525 from sturkmen72:png-spng-test
bc0618b688bde681a8caa8619a55766fc9e6a32f Merge pull request #25582 from fengyuentau:dnn/dump_pbtxt
00440477826a0e7bceaeeaede8351eada5caf38c Merge pull request #25598 from asmorkalov:as/tables_range_check_core
41f08988b4c9756bd528bb6cd0cca0ce104b4edb Merge pull request #25585 from FantasqueX:add-policy-0148
f2d6527a4b429c79a05d6e78dbd6ac6f0cb004f7 Merge pull request #25596 from asmorkalov:as/ci_ubuntu24.04
d29ad2fb719227d2647466b65a928eac6a907c13 Merge pull request #25443 from asmorkalov:as/kleidicv_hal
78ed6de518b60a2e40c74b86ebd76203998d5b50 Merge pull request #25594 from LaurentBerger:I25587
7713c84465b1bbfea112d87ad61312b190a1505d Merge pull request #25297 from CNOCycle:tflite/transpose
76d9f7aaeb8c9ba8aea80bdb155b60c78da1e309 Merge pull request #25591 from LaurentBerger:I25589
68581e7b4af44e374de99117ad886c4273c64714 Merge pull request #25592 from asmorkalov:as/gapi_disable_steaming_again
4d4def2d36e85557c7e57bd678f46a3cb5b8f26b Merge pull request #25590 from asmorkalov:as/restore_highgui_test
47a6ffb73c5f921a6a5941568258fcd2b0825d3b Merge pull request #25561 from Kumataro:fix25560
03507e06b40450244a0489abba8e989d9b337d18 Merge pull request #25518 from alexlyulkov:al/fixed-gemm-openvino
f4e5438401f59386fc2962e2e3521b817797f331 Merge pull request #25566 from asmorkalov:as/moments_documentation
02ad8c284f3f3c50360526a5292af1e74da531a9 Merge pull request #25577 from FantasqueX:fix-zlib-ng-version-parse-1
58e007e18340a1908e13a0071df641c87fdcbee8 Merge pull request #25583 from mshabunin:fix-emd-32bit
3c7e94b27594c8bdfdd9a9541da5f1580246279c Merge pull request #25576 from mshabunin:fix-env-doc
8dc9ff5624fafc4bde69b108ce6e4a468ae3a294 Merge pull request #25551 from Kumataro:fix25550
1d9a4120fa7dbeab6316c255d283c16cd006a35e Merge pull request #25565 from savuor:rv/hal_eq_hist
1f1ba7e40276043fe363102d34a0781920ff29f3 Merge pull request #25563 from asmorkalov:as/HAL_min_max_idx
5bd64e09a360e9fa342fbd951a3635f065bc478e Merge pull request #25554 from savuor:rv/hal_lut
faa259ab344f247c88bd9c4259f77aff3a27aa49 Merge pull request #25553 from asmorkalov:as/HAL_min_max_idx
a9e489f149e24c52edcaaa40c12e7e2adff5d305 Merge pull request #25558 from asmorkalov:as/moments_sanity
ba65d2eb0d83e6c9567a2534f8994cd4484a1635 Merge pull request #25549 from mshabunin:fix-cap-images-noise
8bc90a4bff803e2fa1b74325f25c20d8cdd81afe Merge pull request #25484 from mshabunin:bounding-rect-alignment
26a5730f0d36f547a6eaf39b1e15786ea3f0d039 Merge pull request #25542 from mshabunin:fix-disable-diagram
a4486c0e20950785371cbd74bd5231c15ae71bc2 Merge pull request #25541 from mshabunin:fix-numpy2x-test
70d333d3368809500da029e3f1b07b3e27231409 Merge pull request #25483 from asmorkalov:as/HAL_meanStdDev
c43957cbece8e8fcad32077488edaf99e9ad63c3 Merge pull request #25538 from asmorkalov:as/resnet_optional
75f3f0fac8c48c37a6ec4ff1d4c54ddd09ee32de Merge pull request #25526 from NlSEMONO:patch-1
ac9a858377a51512d9ebf12d89153b024abb7c73 Merge pull request #25524 from alexlyulkov:al/openvino-layers
6e5a53c37a6fd9327b708542a22b84e98d6499ee Merge pull request #25532 from asmorkalov:as/HAL_moments_bug
d9555dbdd1802ee78c104372f6f03f69f9b951fc Merge pull request #25535 from asmorkalov:as/convertfp16_imgcodecs
c539cfd31a69218c4840ba23906dc026d14879fb Merge pull request #25435 from WanliZhong:fcn2onnx
6bf758ecc497bb532dfd1722d87aacf1bd530138 Merge pull request #24782 from FantasqueX:4.x-zlib-ng
7ea4364c63aa8ecde844fc45ff61d2f60c746c17 Merge pull request #25528 from asmorkalov:as/python_hdr_parser_fix
bc95f27e56d89e486c25e76db0573b610bee6496 Merge pull request #25509 from savuor:rv/hal_otsu
ce642e994deebea7c2484a68027703d71b21b3ef Merge pull request #25510 from Kumataro:fix25497
1c9efbd4bed6983b4cfbbb872e259455c7833882 Merge pull request #25523 from Kumataro:fix25522
b5cf4f1483cb3a3884bbec08f289ba8f7b582b91 Merge pull request #25514 from xndcn:patch-1
1f6f092a9d2ba139b825a0dee47a78b0cf6f0319 Merge pull request #25513 from zihaomu:fix_25498
cbf71c380f5e593263ab9bdc9119fb8d81c30de1 Merge pull request #25502 from Kumataro:fix25501
2a2ff5525720d110c9c462cd46e0e0655c7fb0fa Merge pull request #25496 from Kumataro:fix25495
b1e01970ef07c90d407b5c6147c16aeb3d39c7d7 Merge pull request #25308 from dkurt:not_normalized_findHomography
12e2cc9502bc51bb01ed3fdd2f39ce1533c8236e Merge pull request #25491 from savuor:rv/hal_norm_hamming
357b9abaef117665bd2471c495846320aa8877cc Merge pull request #25450 from savuor:rv/svd_perf
8af76698edde60ba475a76a14d56bacc2c022e4b Merge pull request #25493 from vrabaud:tiff
2cd330486ec4597eab49c1575fc4a6603f205a6a Merge pull request #25490 from asmorkalov:as/HAL_moments
7b58ffebd0b52072fa38fcaa7da547c66ef3c4e6 Merge pull request #25486 from asmorkalov:as/gapi_warning_fix
2225b257cfd41a957a2fcfea6f757b09c1d67d11 Merge pull request #25488 from inkredibl:doc-fix-findEssentialMat
ebea65777f30e0090807d410b8ab878517795a32 Merge pull request #24938 from DariaMityagina:icv/dm/add-media-frame-support-to-govbackend
7e56908306934d6e1c337e27a8473d6ea86a8623 Merge pull request #25469 from mshabunin:cpp-emd
5b0843728ebd96196dbeef31cfa81c37f04234d5 Merge pull request #25453 from Kumataro:fix25452
616a913fa2abf499be0d7279aab5f534c3b83d9d Merge pull request #25468 from WanliZhong:change_js_ocvFaceDet_sample
50b6edc9ba8157570a7b147f31b2ea36260c2021 Merge pull request #25321 from dkurt:skip_gapi_test
ae85e516c03d78a1befa5001c6918dfb42529511 Merge pull request #25423 from LuukvandenBent:CalibrateHandEyeDatatypeFix
3f8b56ec499b5eab68873e9d92d04e724cf86bf0 Merge pull request #25459 from Kumataro:investigate_tiff_test

Previous "Merge 4.x": #25460

### CODE REVIEW

**Comment by vrabaud** [comment]
@mshabunin , it seems you are not including all of https://github.com/opencv/opencv/pull/25665 Only missing: `Jptr += 18;` line 405 in 3d/src/fundam.cpp

**Comment by mshabunin** [comment]
Yes, thank you! I added this change and test works better now. Though one case is still failing (LMEDS):
```
[ RUN      ] Calib3d_Homography.Refine
/work/opencv/modules/3d/test/test_homography.cpp:763: Failure
Expected: (sqrt(error.at<double>(0, 0))) <= (method == LMEDS ? 7e-4 : 7e-5), actual: 0.0041369 vs 0.0007
[  FAILED  ] Calib3d_Homography.Refine (2 ms)
```

**Comment by vrabaud** [comment]
That test was added to exercise the Jacobian usage. Indeed, if the LM solver is different, that does not help. Maybe you can try increasing the number of iterations from 10 to 20 to see if that improves the error ?

**Comment by dkurt** [comment]
@vrabaud, seems like it always 10: https://github.com/opencv/opencv/blob/a2d2ea6536df143ca118abc0fbe095e7dadbebd2/modules/3d/src/fundam.cpp#L408

@mshabunin, I think it’s ok to increase threshold for 5.x. As we see, with Jacobian fix the error decreased from 0.0100815 to 0.0041369. The test is quite synthetic and catches Jacobian patch only.

**Comment by mshabunin** [comment]
Thank you! I tried to increase number of iterations, but it did not help. So I've updated the test threshold.

**Comment by mshabunin** [comment]
@alexlyulkov , please take a look at the dnn part of this PR - whether I've correctly merged parts with OpenVINO support (there were discrepancies between #25524 and #25458).

**Comment by asmorkalov** [comment]
minmax.dispatch.cpp changes are missing:
```diff
diff --git a/modules/core/src/minmax.dispatch.cpp b/modules/core/src/minmax.dispatch.cpp
index cd778b0283..6c50cb20b5 100644
--- a/modules/core/src/minmax.dispatch.cpp
+++ b/modules/core/src/minmax.dispatch.cpp
@@ -309,33 +309,6 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
                ocl_minMaxIdx(_src, minVal, maxVal, minIdx, maxIdx, _mask))
 
     Mat src = _src.getMat(), mask = _mask.getMat();
-
-    if (src.dims <= 2)
-    {
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows,
-                 src.depth(), minVal, maxVal, minIdx, maxIdx, mask.data);
-    }
-    else if (src.isContinuous())
-    {
-        int res = cv_hal_minMaxIdx(src.data, 0, (int)src.total()*cn, 1, src.depth(),
-                                   minVal, maxVal, minIdx, maxIdx, mask.data);
-
-        if (res == CV_HAL_ERROR_OK)
-        {
-            // minIdx[0] and minIdx[0] are always 0 for "flatten" version
-            if (minIdx)
-                ofs2idx(src, minIdx[1], minIdx);
-            if (maxIdx)
-                ofs2idx(src, maxIdx[1], maxIdx);
-            return;
-        }
-        else if (res != CV_HAL_ERROR_NOT_IMPLEMENTED)
-        {
-            CV_Error_(cv::Error::StsInternal,
-            ("HAL implementation minMaxIdx ==> " CVAUX_STR(cv_hal_minMaxIdx) " returned %d (0x%08x)", res, res));
-        }
-    }
-
     MinMaxIdxFunc func = getMinMaxIdxFunc(depth);
     CV_Assert( func != 0 );
```

File: modules/3d/src/hal_replacement.hpp [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+   @brief Project points from 3D world space to 2D screen space using rotation and translation matrix and camera intrinsic parameters
+   @param src_data Pointer to 3D points array with coordinates interleaved as X, Y, Z, X, Y, Z,..
+   @param src_step Step between consecutive 3D points
+   @param src_size Amount of points
+   @param dst_data Pointer to resulting projected 2D points with coordinates interleaved as u, v, u, v,..
+   @param dst_step Step between consecutive projected 2D points
+   @param rt_data Pointer to 3x4 array containing rotation-then-translation matrix
+   @param intr_data Pointer to camera intrinsics structure
+*/
+inline int hal_ni_project_points_pinhole64f(const double* src_data, size_t src_step, size_t src_size,
```
Comment (asmorkalov) [inline modules/3d/src/hal_replacement.hpp:148]: The integration part of HAL is missing in the PR. `undistortPoints` was reworked in 5.x. I'll attach solution soon.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: modules/dnn/src/layers/gather_elements_layer.cpp [hunk 2]
```diff
@@ -183,19 +183,13 @@ class GatherElementsLayerImpl CV_FINAL : public GatherElementsLayer
...
-
+        int32_t indicesBoundValue = nodes[0].dynamicCast<InfEngineNgraphNode>()->node.get_shape()[axis];
+        auto indicesBound = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, &indicesBoundValue);
+        auto indices = std::make_shared<ov::op::v0::Convert>(nodes[1].dynamicCast<InfEngineNgraphNode>()->node, ov::element::i32);
         auto indicesNonNegative = std::make_shared<ov::op::v1::Mod>(
-            std::make_shared<ov::op::v1::Add>(nodes[1].dynamicCast<InfEngineNgraphNode>()->node, indicesBound),
+            std::make_shared<ov::op::v1::Add>(indices, indicesBound),
             indicesBound);
+
         auto gatherElements = std::make_shared<ov::op::v6::GatherElements>(
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/dnn/src/layers/gather_elements_layer.cpp]: Restored 5.x version

File: modules/dnn/src/layers/gather_layer.cpp [hunk 3]
```diff
@@ -125,7 +125,7 @@ class GatherLayerImpl CV_FINAL : public GatherLayer
         auto axisNode = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, &m_axis);
         auto gather = std::make_shared<ov::op::v8::Gather>(
             nodes[0].dynamicCast<InfEngineNgraphNode>()->node,
-            nodes[1].dynamicCast<InfEngineNgraphNode>()->node,
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/dnn/src/layers/gather_layer.cpp]: Restored 5.x version

File: modules/dnn/src/layers/scatterND_layer.cpp [hunk 4]
```diff
@@ -250,7 +250,7 @@ class ScatterNDLayerImpl CV_FINAL : public ScatterNDLayer
     {
         auto scatterND = std::make_shared<ov::op::v3::ScatterNDUpdate>(
             nodes[0].dynamicCast<InfEngineNgraphNode>()->node,
-            nodes[1].dynamicCast<InfEngineNgraphNode>()->node,
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/dnn/src/layers/scatterND_layer.cpp]: Restored 5.x version

File: modules/dnn/src/layers/scatter_layer.cpp [hunk 5]
```diff
@@ -243,19 +243,11 @@ class ScatterLayerImpl CV_FINAL : public ScatterLayer
...
-        else
-            CV_Error(Error::StsNotImplemented, "");
-
+        int32_t indicesBoundValue = nodes[0].dynamicCast<InfEngineNgraphNode>()->node.get_shape()[axis];
+        auto indicesBound = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, &indicesBoundValue);
+        auto indices = std::make_shared<ov::op::v0::Convert>(nodes[1].dynamicCast<InfEngineNgraphNode>()->node, ov::element::i32);
         auto indicesNonNegative = std::make_shared<ov::op::v1::Mod>(
-            std::make_shared<ov::op::v1::Add>(nodes[1].dynamicCast<InfEngineNgraphNode>()->node, indicesBound),
+            std::make_shared<ov::op::v1::Add>(indices, indicesBound),
             indicesBound);
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/dnn/src/layers/scatter_layer.cpp]: Restored 5.x version

File: modules/imgproc/test/test_contours_new.cpp [hunk 6]
```diff
@@ -435,6 +435,9 @@ TEST_P(Imgproc_FindContours_Modes2, approx)
         vector<Vec4i> hierarchy;
         findContours(img, contours, hierarchy, mode, method);
 
+        // NOTE: old and new function results might not match when approximation mode is TC89.
+        // Currently this test passes, but might fail for other random data.
+        // See https://github.com/opencv/opencv/issues/25663 for details.
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/imgproc/test/test_contours_new.cpp]: Removed this message

**Comment by mshabunin** [comment]
I've copied that code block to minmax.dispatch.cpp. Perhaps we need to partially backport minmax changes to 4.x to avoid issues in future (is it #24962?).

**Comment by asmorkalov** [comment]
@mshabunin The patch for the `projectPoints` HAL:
```
diff --git a/modules/3d/src/calibration_base.cpp b/modules/3d/src/calibration_base.cpp
index e087111efb..e3886375cd 100644
--- a/modules/3d/src/calibration_base.cpp
+++ b/modules/3d/src/calibration_base.cpp
@@ -41,6 +41,7 @@
 //M*/
 
 #include "precomp.hpp"
+#include "hal_replacement.hpp"
 #include "distortion_model.hpp"
 #include <stdio.h>
 #include <iterator>
@@ -539,20 +540,15 @@ void cv::projectPoints( InputArray _objectPoints,
               (objectPoints.rows == count && objpt_cn*objectPoints.cols == 3) ||
               (objectPoints.rows == 3 && objpt_cn == 1 && objectPoints.cols == count));
 
-    Mat matM(objectPoints.size(), CV_64FC(objpt_cn));
-    objectPoints.convertTo(matM, CV_64F);
     if (objectPoints.rows == 3 && objectPoints.cols == count) {
         Mat temp;
-        transpose(matM, temp);
-        matM = temp;
+        transpose(objectPoints, temp);
+        objectPoints = temp;
     }
 
     CV_Assert( _imagePoints.needed() );
     _imagePoints.create(count, 1, CV_MAKETYPE(objpt_depth, 2), -1, true);
     Mat ipoints = _imagePoints.getMat();
-    ipoints.convertTo(_m, CV_64F);
-    const Point3d* M = matM.ptr<Point3d>();
-    Point2d* m = _m.ptr<Point2d>();
 
     Mat rvec = _rvec.getMat(), tvec = _tvec.getMat();
     if(!((rvec.depth() == CV_32F || rvec.depth() == CV_64F) &&
@@ -656,6 +652,127 @@ void cv::projectPoints( InputArray _objectPoints,
     bool calc_derivatives = dpdr.data || dpdt.data || dpdf.data ||
                             dpdc.data || dpdk.data || dpdo.data;
 
+    if (!calc_derivatives)
+    {
+        if (objpt_depth == CV_32F && ipoints.type() == CV_32F)
+        {
+            float rtMatrix[12] = { (float)R[0], (float)R[1], (float)R[2], (float)t[0],
+                                (float)R[3], (float)R[4], (float)R[5], (float)t[1],
+                                (float)R[6], (float)R[7], (float)R[8], (float)t[2] };
+
+            cv_camera_intrinsics_pinhole_32f intr;
+            intr.fx = (float)fx; intr.fy = (float)fy;
+            intr.cx = (float)cx; intr.cy = (float)cy;
+            intr.amt_k = 0; intr.amt_p = 0; intr.amt_s = 0; intr.use_tau = false;
+
+            switch (ktotal)
+            {
+            case  0: break;
+            case  4: // [k_1, k_2, p_1, p_2]
+                intr.amt_k = 2; intr.amt_p = 2;
+                break;
+            case  5: // [k_1, k_2, p_1, p_2, k_3]
+                intr.amt_k = 3; intr.amt_p = 2;
+                break;
+            case  8: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6]
+                intr.amt_k = 6; intr.amt_p = 2;
+                break;
+            case 12: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6, s_1, s_2, s_3, s_4]
+                intr.amt_k = 6; intr.amt_p = 2; intr.amt_s = 4;
+                break;
+            case 14: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6, s_1, s_2, s_3, s_4, tau_x, tau_y]
+                intr.amt_k = 6; intr.amt_p = 2; intr.amt_s = 4; intr.use_tau = true;
+                break;
+            default:
+                CV_Error(cv::Error::StsInternal, "Wrong number of distortion coefficients");
+            }
+
+            intr.k[0] = (float)k[0];
+            intr.k[1] = (float)k[1];
+            intr.k[2] = (float)k[4];
+            intr.k[3] = (float)k[5];
+            intr.k[4] = (float)k[6];
+            intr.k[5] = (float)k[7];
+
+            intr.p[0] = (float)k[2];
+            intr.p[1] = (float)k[3];
+
+            for (int ctr = 0; ctr < 4; ctr++)
+            {
+                intr.s[ctr] = (float)k[8+ctr];
+            }
+
+            intr.tau_x = (float)k[12];
+            intr.tau_y = (float)k[13];
+
+            CALL_HAL(projectPoints, cv_hal_project_points_pinhole32f,
+                     (float*)objectPoints.data, objectPoints.step, count,
+                     (float*)ipoints.data, ipoints.step, rtMatrix, &intr);
+        }
+
+        if (objpt_depth == CV_64F && ipoints.type() == CV_64F)
+        {
+            double rtMatrix[12] = { R[0], R[1], R[2], t[0],
+                                    R[3], R[4], R[5], t[1],
+                                    R[6], R[7], R[8], t[2] };
+
+            cv_camera_intrinsics_pinhole_64f intr;
+            intr.fx = fx; intr.fy = fy;
+            intr.cx = cx; intr.cy = cy;
+            intr.amt_k = 0; intr.amt_p = 0; intr.amt_s = 0; intr.use_tau = false;
+
+            switch (ktotal)
+            {
+            case  0: break;
+            case  4: // [k_1, k_2, p_1, p_2]
+                intr.amt_k = 2; intr.amt_p = 2;
+                break;
+            case  5: // [k_1, k_2, p_1, p_2, k_3]
+                intr.amt_k = 3; intr.amt_p = 2;
+                break;
+            case  8: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6]
+                intr.amt_k = 6; intr.amt_p = 2;
+                break;
+            case 12: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6, s_1, s_2, s_3, s_4]
+                intr.amt_k = 6; intr.amt_p = 2; intr.amt_s = 4;
+                break;
+            case 14: // [k_1, k_2, p_1, p_2, k_3, k_4, k_5, k_6, s_1, s_2, s_3, s_4, tau_x, tau_y]
+                intr.amt_k = 6; intr.amt_p = 2; intr.amt_s = 4; intr.use_tau = true;
+                break;
+            default:
+                CV_Error(cv::Error::StsInternal, "Wrong number of distortion coefficients");
+            }
+
+            intr.k[0] = k[0];
+            intr.k[1] = k[1];
+            intr.k[2] = k[4];
+            intr.k[3] = k[5];
+            intr.k[4] = k[6];
+            intr.k[5] = k[7];
+
+            intr.p[0] = k[2];
+            intr.p[1] = k[3];
+
+            for (int ctr = 0; ctr < 4; ctr++)
+            {
+                intr.s[ctr] = k[8+ctr];
+            }
+
+            intr.tau_x = k[12];
+            intr.tau_y = k[13];
+
+            CALL_HAL(projectPoints, cv_hal_project_points_pinhole64f,
+                     (double*)objectPoints.data, objectPoints.step, count,
+                     (double*)ipoints.data, ipoints.step, rtMatrix, &intr);
+        }
+    }
+
+    Mat matM(objectPoints.size(), CV_64FC(objpt_cn));
+    objectPoints.convertTo(matM, CV_64F);
+    ipoints.convertTo(_m, CV_64F);
+    const Point3d* M = matM.ptr<Point3d>();
+    Point2d* m = _m.ptr<Point2d>();
+
     for( i = 0; i < count; i++ )
     {
         double X = M[i].x, Y = M[i].y, Z = M[i].z;
```
@savuor please review the proposed change.

**Comment by mshabunin** [comment]
@asmorkalov , I've added fix for `projectPoints`.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**